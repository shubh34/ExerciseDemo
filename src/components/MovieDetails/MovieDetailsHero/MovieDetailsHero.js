import React from 'react';
import './MovieDetailsHero.scss';
import PropTypes from 'prop-types';


export const MovieDetailsHero = (props) => {
	const { heroUrl, children } = props;
	const style = {
		backgroundImage: `url(${heroUrl})`,
	};
	return (
		<div className="movie-details_hero" style={style}>
			{children}
		</div>
	);
};

MovieDetailsHero.propTypes = {
	heroUrl: PropTypes.string.isRequired,
	children: PropTypes.element.isRequired,
};
