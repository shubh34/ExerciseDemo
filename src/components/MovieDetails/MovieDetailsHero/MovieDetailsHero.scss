.movie-details_hero {
    position: relative;
    background-size: 215%;
    height: 550px;
    background-image: url(https://prod3-kraken-cloudfront.rakuten.tv/images/placeholders/hero-1bbf9a928a.jpg);
    background-color: #000;
    background-repeat: no-repeat;
    background-position: center top;
    will-change: background-position;
    @media screen and (min-width: 560px) {
        background-size: cover;
    }
    @media screen and (min-width: 1140px) {
        height: 620px;
    }
}
