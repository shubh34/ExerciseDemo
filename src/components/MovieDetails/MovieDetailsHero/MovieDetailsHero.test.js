

import React from 'react';
import { shallow } from 'enzyme';

import { MovieDetailsHero } from './MovieDetailsHero';


const setUp = (container = shallow) => {
	const wrapper = container(
		<MovieDetailsHero heroUrl="test.jpg">
			<div>Wrapper</div>
		</MovieDetailsHero>,
	);
	return { wrapper };
};

describe('MovieDetailsHero', () => {
	it('should render children in wrapper', () => {
		const { wrapper } = setUp();
		expect(wrapper).toMatchSnapshot();
	});
});
