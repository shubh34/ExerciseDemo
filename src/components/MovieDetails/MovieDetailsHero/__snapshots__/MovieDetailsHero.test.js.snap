// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieDetailsHero should render children in wrapper 1`] = `
<div
  className="movie-details_hero"
  style={
    Object {
      "backgroundImage": "url(test.jpg)",
    }
  }
>
  <div>
    Wrapper
  </div>
</div>
`;
